server/xcode/xcode_darwin.go

```diff
@@ -168,3 +168,25 @@ func stringToCFString(gostr string) C.CFStringRef {
 func stringFromCFString(cfStr C.CFStringRef) string {
 	return C.GoString(C.CFStringGetCStringPtr(cfStr, C.kCFStringEncodingUTF8))
 }
+
+func sdksString(sdks map[string]string) string {
+	availableSDKs := make([]string, 0, len(sdks))
+	i := 0
+	for sdk, _ := range sdks {
+		availableSDKs[i] = sdk
+		i++
+	}
```

Comment:
[make](https://pkg.go.dev/builtin#make)() (for slices) accepts a length and capacity as the second and third arg.

Typically you want to set one or both if you know the length of slice you'll need ahead of time so that you avoid the extra allocations that would happen to dynamically resize the slice. (Length obviously 'includes' capacity)

So the 3 most common patterns are:

1) no idea how big the slice is, just make a 0 length slice and append to it:
slice := make([]string, 0)
for thing in blah {
  slice = append(slice, thing)
}

2) you know how big the slice is, and you have an index that increases by one for each item
slice := make([]int, count)
for i := 0; i < count; i++ {
  slice[i] = i * i
}

3) you know how big the slice (or roughly how big it is) is but you just want to put items in it
slice := make([]string, 0, count)
for k, v in myMap {
  if v == nil {
    continue
  }
  slice = append(slice, k)
} 


Generally it is clearest to use 3) so you don't need to keep track of i. And since you set the slice capacity you're avoiding the extra allocations already.

So for these loops, here and below, can we remove "i" and use append instead?

Reply:
Based on https://stackoverflow.com/a/27848197 it seemed best to use `i`. I forgot to adjust the `make()`. I can adjust it to use append though.

Reply:
(Because of "Although setting the capacity eliminates reallocations, append still has to do extra work to check if you've reached capacity on each append.")

Reply:
Yeah, this is correct (it is slightly faster to use an index like this), but it comes at the cost of having to declare the extra counter variable.

I think for stuff like this (displaying a convenience string to the user) I would prefer the less code / more readable / simpler to refactor version.